### PR TITLE
Feature/MDS-1967 Create Party Business Roles, Add Inspectors; MDS-1957 Party Select uses search endpoint.

### PR DESCRIPTION
**Pull request by nayr974**
This PR adds the concept of a party business role, of which currently there will only be inspector. It refactors Variances to use parties versus core users. 

Additionally, the Party Select component is modified to use the /search endpoint, which has better logic for searching parties.

### CODE REVIEW

File: python-backend/app/api/mines/variances/resources/variance.py [hunk 1]
```diff
@@ -85,16 +83,16 @@ def put(self, mine_guid, variance_guid):
...
-            core_user = CoreUser.find_by_core_user_guid(core_user_guid)
-            if not core_user:
+        inspector_party_guid = data.get('inspector_party_guid')
+        if inspector_party_guid:
+            inspector = Party.find_by_party_guid(inspector_party_guid)
+            if not inspector:
                 raise BadRequest('Unable to find new inspector.')
 
-            variance.inspector_id = core_user.core_user_id
+            variance.inspector_party_guid = inspector_party_guid
```
Comment (Jsyro) [inline python-backend/app/api/mines/variances/resources/variance.py]: Should we validate this person is infact an inspector?

File: frontend/src/tests/components/modalContent/__snapshots__/AddVarianceModal.spec.js.snap [hunk 2]
```diff
@@ -147,23 +117,7 @@ ShallowWrapper {
...
-                  "department": "Mines Health, Safety & Enforcement Division",
-                  "title": "Geomatics Specialist",
-                  "username": "IDIR\\\\BLAH",
-                },
-                "last_logon": null,
-                "phone_no": "555-555-5556",
-              },
-            ]
-          }
+          inspectors={undefined}
```
Comment (Jsyro) [inline frontend/src/tests/components/modalContent/__snapshots__/AddVarianceModal.spec.js.snap]: this smells? change other test to use your new mock?

File: python-backend/app/api/parties/party_appt/models/party_business_role_appt.py [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+    __tablename__ = "party_business_role_appt"
+    # Columns
+    party_business_role_appt_id = db.Column(
+        db.Integer, primary_key=True, server_default=FetchedValue())
+    party_business_role_code = db.Column(
+        db.String(32), db.ForeignKey('party_business_role_code.party_business_role_code'))
+    party_guid = db.Column(UUID(as_uuid=True), db.ForeignKey('party.party_guid'))
+    start_date = db.Column(db.DateTime)
+    end_date = db.Column(db.DateTime)
+    deleted_ind = db.Column(db.String(60), server_default=FetchedValue())
```
Comment (Jsyro) [inline python-backend/app/api/parties/party_appt/models/party_business_role_appt.py]: string?

File: python-backend/app/api/mines/variances/resources/variance.py (hunk 1 above)
Comment (nayr974) [reply python-backend/app/api/mines/variances/resources/variance.py]: Added a check.

File: frontend/src/tests/components/modalContent/__snapshots__/AddVarianceModal.spec.js.snap (hunk 2 above)
Comment (nayr974) [reply frontend/src/tests/components/modalContent/__snapshots__/AddVarianceModal.spec.js.snap]: Set to mock data.

File: python-backend/app/api/parties/party_appt/models/party_business_role_appt.py (hunk 3 above)
Comment (nayr974) [reply python-backend/app/api/parties/party_appt/models/party_business_role_appt.py]: No...........

**Pull request merged**